### PR TITLE
Update FastDEC after ACSet updates

### PR DESCRIPTION
**Pull request by GeorgeR227**
This is meant to replace views in the FastDEC module with regular ACSet accessing. I've also taken the liberty of cleaning up some strange logic.

### CODE REVIEW

**Comment by GeorgeR227** [comment]
A few benchmarks of note. Overall performance otherwise is preserved.

### Old:
Operator: Geometric Hodge
Variant: New Form-1, [0.935041 ms, 2.212816 MB]
Variant: Old Form-1, [53.808375 ms, 54.740096 MB]

Operator: Wedge Product
Variant: New Form-0, Form-1, [0.052583 ms, 0.308864 MB]
Variant: New Form-0, Form-2, [0.339291 ms, 0.415056 MB]
Variant: New Form-1, Form-1, [0.1625 ms, 0.35056 MB]
Variant: Old Form-0, Form-1, [8.164167 ms, 8.05008 MB]
Variant: Old Form-0, Form-2, [46.512188 ms, 83.6008 MB]
Variant: Old Form-1, Form-1, [50.183625 ms, 56.1576 MB]

### New:
Operator: Geometric Hodge
Variant: New Form-1, [0.9962295 ms, 1.8428 MB]

Operator: Wedge Product
Variant: New Form-0, Form-1, [0.053959 ms, 0.308864 MB]
Variant: New Form-0, Form-2, [0.257666 ms, 0.409744 MB]
Variant: New Form-1, Form-1, [0.154208 ms, 0.35056 MB]

**Comment by GeorgeR227** [comment]
```julia
Test Summary: | Pass  Total   Time
CUDA          |  139    139  41.7s
     Testing CombinatorialSpaces tests passed 

shell> git log -1
commit 67ad092ad6615e66994ce3956d7a65657fdf3684 (HEAD -> gr/clean-fast-dec, origin/gr/clean-fast-dec)
Author: GeorgeR227 <78235421+GeorgeR227@users.noreply.github.com>
Date:   Tue Jun 4 15:32:09 2024 -0400

    Added some doc strings.
```

**Comment by lukem12345** [comment]
Can you edit your comment to use the commit at which you ran those benchmarks

File: src/FastDEC.jl [hunk 1]
```diff
@@ -485,20 +477,18 @@ function dec_p_derivbound(::Type{Val{1}}, sd::HasDeltaSet; transpose::Bool=false
...
 
-    v1_list = @view sd[:D_∂v1]
-    dual_lengths = @view sd[:dual_length]
-
-    for (d_edge_idx, v1) in enumerate(v1_list)
-        if (1 <= v1 <= num_v_sd)
-            hodge_diag_0[v1] += dual_lengths[d_edge_idx]
-        end
+    for d_edge_idx in parts(sd, :DualE)
+      v1 = sd[d_edge_idx, :D_∂v1]
```
Comment (lukem12345) [inline src/FastDEC.jl:488]: Checking that this is the dual vertex that is at the center of a primal vertex may just "fall out" of using our accessors in a certain order. So the following reliance on checking if it is in 1:nv(sd) may be unnecessary.

Comment (lukem12345) [reply src/FastDEC.jl:488]: Further, this may be faster using `@inbounds`

**Review by lukem12345 — APPROVED** [review]

**Pull request merged**